From 7aa409a6d110d24e4c94ec6e7ce3cacb0e7e26bd Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 12 Sep 2021 15:21:19 +0300
Subject: [PATCH] [SW-55766] gaudi2: pci: moving iatu init to arch common code

Both Arc and Arm are required to initialize the iATU so the
initialization function better be in the common code for both
architectures.

Change-Id: Idd021d6c09cf33ab458d2ab4f9029d8bf8a761aa
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index 89c6056..4d3b111 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -38,42 +38,3 @@
 
 	return !memcmp(hash, desc->chksum, ARRAY_SIZE(hash));
 }
-
-void hl_pci_iatu_init(void)
-{
-	struct hl_pci_inbound_region inbound_region;
-	struct hl_pci_outbound_region outbound_region;
-
-	/* TODO: implement hl_pcie_disable_access() for gaudi2 */
-	hl_pcie_disable_access();
-
-	/* Inbound Region 0 - Bar 0 - Point to CFG */
-	inbound_region.mode = PCI_ADDRESS_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.offset_in_bar = 0;
-	inbound_region.addr = STM_FLASH_BASE;
-	inbound_region.size = CFG_REGION_SIZE;
-	hl_pci_set_inbound_region(0, &inbound_region);
-
-	/* Inbound Region 1 - Bar 0 - Point to BAR0_RESERVED + SRAM */
-	inbound_region.mode = PCI_ADDRESS_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.offset_in_bar = CFG_REGION_SIZE;
-	inbound_region.addr = BAR0_RSRVD_BASE;
-	inbound_region.size = BAR0_RSRVD_SIZE + SRAM_SIZE;
-	hl_pci_set_inbound_region(1, &inbound_region);
-
-	/* Inbound Region 2 - Bar 4 - Point to DRAM */
-	inbound_region.mode = PCI_BAR_MATCH_MODE;
-	inbound_region.bar = PCI_BAR_ID_4;
-	inbound_region.addr = DRAM_BASE;
-	hl_pci_set_inbound_region(2, &inbound_region);
-
-	/* Outbound Region 0 - Point to Host */
-	outbound_region.addr = HOST_PHYS_BASE_0;
-	outbound_region.size = HOST_PHYS_SIZE_0;
-	hl_pci_set_outbound_region(&outbound_region);
-
-	/* TODO: implement hl_pcie_enable_access() for gaudi2 */
-	hl_pcie_enable_access();
-}
diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 81ada1c..54a3ba7 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -8,10 +8,12 @@
 #include <gaudi2_blocks_linux_driver.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
+#include <address_space.h>
 
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
+#include <pci.h>
 
 static struct hl_pll_cfg gaudi2_pll_cfg[] = {
 	{
@@ -144,3 +146,42 @@
 
 	return factor > 2 ? factor : 2;
 }
+
+void hl_pci_iatu_init(void)
+{
+	struct hl_pci_inbound_region inbound_region;
+	struct hl_pci_outbound_region outbound_region;
+
+	/* TODO: implement hl_pcie_disable_access() for gaudi2 */
+	hl_pcie_disable_access();
+
+	/* Inbound Region 0 - Bar 0 - Point to CFG */
+	inbound_region.mode = PCI_ADDRESS_MATCH_MODE;
+	inbound_region.bar = PCI_BAR_ID_0;
+	inbound_region.offset_in_bar = 0;
+	inbound_region.addr = STM_FLASH_BASE;
+	inbound_region.size = CFG_REGION_SIZE;
+	hl_pci_set_inbound_region(0, &inbound_region);
+
+	/* Inbound Region 1 - Bar 0 - Point to BAR0_RESERVED + SRAM */
+	inbound_region.mode = PCI_ADDRESS_MATCH_MODE;
+	inbound_region.bar = PCI_BAR_ID_0;
+	inbound_region.offset_in_bar = CFG_REGION_SIZE;
+	inbound_region.addr = BAR0_RSRVD_BASE;
+	inbound_region.size = BAR0_RSRVD_SIZE + SRAM_SIZE;
+	hl_pci_set_inbound_region(1, &inbound_region);
+
+	/* Inbound Region 2 - Bar 4 - Point to DRAM */
+	inbound_region.mode = PCI_BAR_MATCH_MODE;
+	inbound_region.bar = PCI_BAR_ID_4;
+	inbound_region.addr = DRAM_BASE;
+	hl_pci_set_inbound_region(2, &inbound_region);
+
+	/* Outbound Region 0 - Point to Host */
+	outbound_region.addr = HOST_PHYS_BASE_0;
+	outbound_region.size = HOST_PHYS_SIZE_0;
+	hl_pci_set_outbound_region(&outbound_region);
+
+	/* TODO: implement hl_pcie_enable_access() for gaudi2 */
+	hl_pcie_enable_access();
+}
